catalog/core/catalog-core-api/src/main/java/ddf/catalog/filter/SimpleFilterDelegate.java

```diff
@@ -0,0 +1,679 @@
+        return propertyIsLessThanOrEqualTo(propertyName,
+                literal,
+                Object.class,
+                ComparisonPropertyOperation.IS_LESS_OR_EQUAL_TO);
+    }
+
+    public <S> T propertyIsLessThanOrEqualTo(String propertyName, S literal, Class<S> literalClass,
+            ComparisonPropertyOperation operation) {
+        return comparisonOperation(propertyName, literal, literalClass, operation);
+    }
+
+    // PropertyIsBetween
+
+    public T propertyIsBetween(String propertyName, String lowerBoundary, String upperBoundary) {
+        return propertyIsBetween(propertyName,
+                lowerBoundary,
+                upperBoundary,
+                String.class,
+                ComparisonPropertyOperation.IS_BETWEEN);
+    }
+
+    public T propertyIsBetween(String propertyName, Date lowerBoundary, Date upperBoundary) {
+        return propertyIsBetween(propertyName,
+                lowerBoundary,
+                upperBoundary,
+                Date.class,
+                ComparisonPropertyOperation.IS_BETWEEN);
+    }
+
+    public T propertyIsBetween(String propertyName, int lowerBoundary, int upperBoundary) {
+        return propertyIsBetween(propertyName, (Number) lowerBoundary, (Number) upperBoundary);
+    }
+
+    public T propertyIsBetween(String propertyName, short lowerBoundary, short upperBoundary) {
+        return propertyIsBetween(propertyName, (Number) lowerBoundary, (Number) upperBoundary);
+    }
+
+    public T propertyIsBetween(String propertyName, long lowerBoundary, long upperBoundary) {
+        return propertyIsBetween(propertyName, (Number) lowerBoundary, (Number) upperBoundary);
+    }
+
+    public T propertyIsBetween(String propertyName, float lowerBoundary, float upperBoundary) {
+        return propertyIsBetween(propertyName, (Number) lowerBoundary, (Number) upperBoundary);
+    }
+
+    public T propertyIsBetween(String propertyName, double lowerBoundary, double upperBoundary) {
+        return propertyIsBetween(propertyName, (Number) lowerBoundary, (Number) upperBoundary);
+    }
+
+    public T propertyIsBetween(String propertyName, Number lowerBoundary, Number upperBoundary) {
+        return propertyIsBetween(propertyName,
+                lowerBoundary,
+                upperBoundary,
+                Number.class,
+                ComparisonPropertyOperation.IS_BETWEEN);
+    }
+
+    public T propertyIsBetween(String propertyName, Object lowerBoundary, Object upperBoundary) {
+        return propertyIsBetween(propertyName,
+                lowerBoundary,
+                upperBoundary,
+                Object.class,
+                ComparisonPropertyOperation.IS_BETWEEN);
+    }
+
+    public <S> T propertyIsBetween(String propertyName, S lowerBoundary, S upperBoundary,
+            Class<S> literalClass, ComparisonPropertyOperation operation) {
+        return comparisonOperation(propertyName, lowerBoundary, literalClass, operation);
```

Comment:
Only lower bound is getting passed.  Consider using `Range` from Guava to pass the lower and upper boundary to `comparisonOperation`